core/lib/src/request/request.rs

```diff
@@ -386,6 +386,84 @@ impl<'r> Request<'r> {
             })
     }
 
+    /// Returns the [`ProxyProto`] associated with the current request.
+    ///
+    /// The value is determined by inspecting the header named
+    /// [`proxy_proto_header`](crate::Config::proxy_proto_header), if
+    /// configured. If parameter isn't configured or the request doesn't contain
+    /// a header named as indicated, this method returns `None`.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use rocket::http::{Header, ProxyProto};
+    ///
+    /// # let c = rocket::local::blocking::Client::debug_with(vec![]).unwrap();
+    /// # let req = c.get("/");
+    /// // By default, no `proxy_proto_header` is configured.
+    /// let req = req.header(Header::new("x-forwarded-proto", "https"));
+    /// assert_eq!(req.proxy_proto(), None);
+    ///
+    /// // We can configured one by setting the `proxy_proto_header` parameter.
+    /// // Here we set it to `x-forwarded-proto`, considered de-facto standard.
+    /// # let figment = rocket::figment::Figment::from(rocket::Config::debug_default());
+    /// let figment = figment.merge(("proxy_proto_header", "x-forwarded-proto"));
+    /// # let c = rocket::local::blocking::Client::debug(rocket::custom(figment)).unwrap();
+    /// # let req = c.get("/");
+    /// let req = req.header(Header::new("x-forwarded-proto", "https"));
+    /// assert_eq!(req.proxy_proto(), Some(ProxyProto::Https));
+    ///
+    /// # let req = c.get("/");
+    /// let req = req.header(Header::new("x-forwarded-proto", "http"));
+    /// assert_eq!(req.proxy_proto(), Some(ProxyProto::Http));
+    ///
+    /// # let req = c.get("/");
+    /// let req = req.header(Header::new("x-forwarded-proto", "xproto"));
+    /// assert_eq!(req.proxy_proto(), Some(ProxyProto::Unknown("xproto".into())));
+    /// ```
+    pub fn proxy_proto(&self) -> Option<ProxyProto<'_>> {
+        self.rocket().config
+            .proxy_proto_header
+            .as_ref()
+            .and_then(|header| self.headers().get_one(header.as_str()))
+            .map(ProxyProto::from)
```

Comment:
```suggestion
        self.rocket()
            .config
            .proxy_proto_header
            .as_ref()
            .and_then(|header| self.headers().get_one(header.as_str()))
            .map(ProxyProto::from)
```

Reply:
Already fixed in 678b7c1f8dfe59d43a6295c0221d102ba720d520